internal/service/rds/engine_version_data_source.go

```diff
@@ -136,6 +149,18 @@ func dataSourceEngineVersionRead(d *schema.ResourceData, meta interface{}) error
 		input.Engine = aws.String(v.(string))
 	}
 
+	if v, ok := d.GetOk("default_only"); ok {
+		input.DefaultOnly = aws.Bool(v.(bool))
```

Comment:
While updating the tests, I noticed that `DefaultOnly` is already being set below separately if neither `version` nor `preferred_versions` are set. Let me know if you'd like me to restructure this code to combine the two sections somehow.

I could see checking explicitly first for `default_only`, then having an `else if` with the existing checks to set `DefaultOnly` implicitly when `version` and `preferred_versions` are unset. Or I could see only checking `default_only` if one of `version` or `preferred_versions` are set. Another thing to consider is whether it makes sense to return an error if `default_only = false` is set explicitly without setting `version` or `preferred_versions`.

Reply:
```diff
diff --git a/internal/service/rds/engine_version_data_source.go b/internal/service/rds/engine_version_data_source.go
index 2a2d6206e1..9c8daba4c7 100644
--- a/internal/service/rds/engine_version_data_source.go
+++ b/internal/service/rds/engine_version_data_source.go
@@ -149,10 +149,6 @@ func dataSourceEngineVersionRead(d *schema.ResourceData, meta interface{}) error
 		input.Engine = aws.String(v.(string))
 	}
 
-	if v, ok := d.GetOk("default_only"); ok {
-		input.DefaultOnly = aws.Bool(v.(bool))
-	}
-
 	if v, ok := d.GetOk("filter"); ok {
 		input.Filters = namevaluesfilters.New(v.(*schema.Set)).RDSFilters()
 	}
@@ -169,7 +165,9 @@ func dataSourceEngineVersionRead(d *schema.ResourceData, meta interface{}) error
 		input.EngineVersion = aws.String(v.(string))
 	}
 
-	if _, ok := d.GetOk("version"); !ok {
+	if v, ok := d.GetOk("default_only"); ok {
+		input.DefaultOnly = aws.Bool(v.(bool))
+	} else if _, ok := d.GetOk("version"); !ok {
 		if _, ok := d.GetOk("preferred_versions"); !ok {
 			input.DefaultOnly = aws.Bool(true)
 		}
```

I feel like this is the right change to make. If a user explicitly sets `default_only`, the flag should not be set to a potentially different value just because `version` and `preferred_versions` are not set. If they set `default_only = false` without setting the other attributes in a way that will guarantee that the lookup only returns a single version, then it should be expected to fail.

Reply:
I ended up making this change since it makes more sense to me. I can always change it back or to some other approach if that's what the eventual reviewer prefers.